### ISSUE TITLE
Proposal for new api docs structure

### ISSUE THREAD

**Issue by jonathanlukas** [camunda/camunda-docs#4353]
Slack thread: https://camunda.slack.com/archives/C069H0BDB3R/p1727264930460509

Motivation: With the introduction of a v2 api, we will need to guide current users to the new endpoints.

As our client libs should have this implemented by default, the documentation should state this as well.

The docs structure should clarify the api versions:

```
* APIs
  * Automation
    * v1
      * Operate API
      * Tasklist API
      * Optimize API
      * Zeebe REST API
      * Zeebe gRPC API <- this is technically not accurate, but would help raise understanding
      * Identity REST API <- this is interely missing as of now, maybe can be skipped at all
    * v2
      * Camunda REST API
      * Camunda gRPC API <- this is technically not accurate, but would help raise understanding
  * Web Modeler REST API
  * Management REST API
```

Next, each documented endpoint from v1 should point to the equivalent in v2.

This makes it more clear WHY APIs are actually deprecated and that features are not lost, but ported over to a new, better-designed and unified api which will improve devex a lot.

**Comment by jonathanlukas** [comment]
With client libs, I refer to the operate and tasklist client. I maintain them and will create a v2 implementation that is basically just a wrapper for the camunda client. This will also lower the hurdle for migration.

I would see the actual migration guidance inside each v1 api (best case for each request with a link to the new api method, or with an interactive request migration tool). 

My assumption has always been that we will not just remove an api in the next release, just marking them as deprecated.

To have further guidance, we can:

* inside the "automation" bullet point, move the v2 over the v1
* indicate that v1 should be avoided (call it v1 (deprecated)?) -> follow up question: What would that mean for the Optimize API?

**Comment by akeller** [comment]
@christinaausley @pepopowitz @aleksander-dytko can I get your thoughts here on this proposed structure?

**Comment by pepopowitz** [comment]
I definitely agree that we should organize the API docs to encourage the preferred path for users. 

For someone who has already run into the v1/v2 situation, they should have no problem finding what they're looking for with this proposed layout. I'm not sure if organizing by version number is clear to a user who is interacting with our APIs for the first time, though. I suppose they might just assume to look at the latest. I think that I would look at it and wish that I had some guidance on which version I needed, and eventually settle on v2.

Another issue related to grouping by versions is that the version doesn't really tell you which are the "current" APIs. You can see this in the proposed structure, as the Web Modeler and Management APIs sit outside of a version folder. You can also see it with the inline "this is not technically accurate" comments, I think. And I can also imagine a future where one preferred API moves to v3, and we have v1, v2, and v3 APIs that are all preferred, but scattered across this structure.

## My counter-proposal

I think I prefer something not much different than what we currently have, but with our not-preferred-but-not-deprecated APIs quarantined into their own section. (If this state of API exists. I don't actually know the status of all APIs, but I see Tasklist API not in deprecated in docs.camunda.io, so I think this means that it does.) Something like this: 

```
- APIs
  - Administration API (REST)
  - Administration API (Self-Managed)
  - Camunda 8 API (REST)
  - ....
  - Legacy APIs
    - Tasklist API (REST)
    - ...
  - Deprecated APIs
    - Tasklist API (GraphQL)
    - ...
```

We could _explicitly_ call out the preferred APIs by giving them a folder also, but I don't feel that's necessary. Having all our favorites as direct children of "APIs" accomplishes that, IMO. 

## But wait, there's more

Beyond just organization, I really think we could benefit from some more documentation here. At the very least, a doc that sits first under the APIs section (or a page for the APIs section itself) with a more appropriate name than "Making sense of the Camunda APIs", but accomplishes the same goal. We have so many APIs, and I think it's intimidating and confusing to come into our API docs if I don't have full knowledge of what each is for. We could use this doc to explain scenarios where each API is useful, what it means for an API to be considered "legacy," and what it means for an API to be considered "deprecated." We may also decide to put documents like this inside each "avoid" folder (Legacy & Deprecated), but at the very least, one doc at the top of the tree please.

We might already have a doc like this in a migration or best practices section, but I'm unaware of it.

**Comment by jonathanlukas** [comment]
Also a good approach. So you don't like the explicit api versioning. My assumption would always be that there is only one **current** api for one functionality (and this api version does not need to be the latest one as we still have v1 web modeler and administration api for example).

Plus, the legacy section makes total sense as this is a clear guidance.

**Comment by christinaausley** [comment]
> Beyond just organization, I really think we could benefit from some more documentation here. At the very least, a doc that sits first under the APIs section (or a page for the APIs section itself) with a more appropriate name than "Making sense of the Camunda APIs", but accomplishes the same goal. We have so many APIs, and I think it's intimidating and confusing to come into our API docs if I don't have full knowledge of what each is for. We could use this doc to explain scenarios where each API is useful, what it means for an API to be considered "legacy," and what it means for an API to be considered "deprecated." We may also decide to put documents like this inside each "avoid" folder (Legacy & Deprecated), but at the very least, one doc at the top of the tree please.

I really like this idea, and I'd be happy to put this together, if we'd like.

Additionally, I have [this open issue](https://github.com/camunda/developer-experience/issues/431) which semi-relates to this restructure -- how would we feel about removing (REST) from the sidebar titles, particularly given GraphQL will be removed with https://github.com/camunda/camunda-docs/pull/4649?

**Comment by pepopowitz** [comment]
> Additionally, I have [this open issue](https://github.com/camunda/developer-experience/issues/431) which semi-relates to this restructure -- how would we feel about removing (REST) from the sidebar titles, particularly given GraphQL will be removed with #4649?

I am a ➕ to removing REST from the titles.

**Comment by aleksander-dytko** [comment]
I agree with the direction that @pepopowitz proposed. In my mind, the APIs section for 8.7 could look like:

```
- APIs
  - Camunda 8 API
  - Administration API (SaaS)
  - Administration API (Self-Managed)
  - Zeebe gPRC -> We would have a streaming API over here that will be maintained 
  - Optimize API 
  - WebModeler API 
  - Deprecated APIs
    - Tasklist API 
    - Operate API
    - Zeebe REST API
    - Zeebe gRPC API
```

In every deprecated API endpoint, we should have at least a link to a new (supported) endpoint from C8 API. This way, developers can easily understand what is the new endpoint that they should use.

**Comment by akeller** [comment]
I've added this to the Leveling Up Documentation initiative, but I think we need to agree on next steps. 

* Sidebar order - alphabetical, C8 API first then alphabetical, some other order (I vote C8 API first then alphabetical)
* Removing REST (REST is implied for most API) - I'm ok with this, as long as we keep gRPC clearly
  * Instead of "Zeebe gRPC" in both sections, can we call this "Zeebe Streaming gRPC" or something similar that clearly indicates only a small subset of that remains while the rest is deprecated? 

@aleksander-dytko as our PM in this space, what would you like to see? And what could we achieve today (meaning for 8.7 or 8.8)?

**Comment by akeller** [comment]
> I've added this to the Leveling Up Documentation initiative, but I think we need to agree on next steps.
> 
> * Sidebar order - alphabetical, C8 API first then alphabetical, some other order (I vote C8 API first then alphabetical)
> * Removing REST (REST is implied for most API) - I'm ok with this, as long as we keep gRPC clearly
>   
>   * Instead of "Zeebe gRPC" in both sections, can we call this "Zeebe Streaming gRPC" or something similar that clearly indicates only a small subset of that remains while the rest is deprecated?
> 
> [@aleksander-dytko](https://github.com/aleksander-dytko) as our PM in this space, what would you like to see? And what could we achieve today (meaning for 8.7 or 8.8)?

@aleksander-dytko do you agree with next steps here to deliver in 8.8?

**Comment by aleksander-dytko** [comment]
@akeller yes, the proposal makes sense to me. 

-> Let's have Orchestration Cluster API as the first one, then other APIs
-> Keep `Zeebe Streaming gRPC` as supported. Move `Zeebe gRPC` to deprecated section

**Comment by akeller** [comment]
@aleksander-dytko when you say the "Zeebe Streaming gRPC" you just mean https://docs.camunda.io/docs/next/apis-tools/zeebe-api/gateway-service/#streamactivatedjobs-rpc?
